### PR TITLE
feat: allow adding more trusted principals to task role

### PR DESCRIPTION
**Pull request by parmouraly**
## Description
Extra input variable to allow module users to optionally define extra principals that can assume the task role.
Apart from the main module code and documentation, the example is also updated to showcase how the new variable can be used.

## Motivation and Context
At the moment, the only trusted entity to assume the task role is the absolute minimum `ecs-tasks.amazonaws.com`.
However sometimes it's convenient to allow adding more principals.
For example in order to allow easy SSM access into the Fargate task, `ssm.amazonaws.com` should be trusted too.

## Breaking Changes
No breaking changes. Just adding an optional input variable.

## How Has This Been Tested?
From my project downstream, I have referenced this PR branch and applied my infrastructure. It works as expected.

### CODE REVIEW

**Comment by parmouraly** [comment]
Hi @antonbabenko 👋 
I would like to test this PR before I mark it as _ready for review_.
At the moment I am using this module with 
`source  = "terraform-aws-modules/atlantis/aws"`

Is it possible for me to change this `source` path so I can reference my branch?
I have tried unsuccessfully something like
`source = "git::ssh://github.com/parmouraly/terraform-aws-atlantis.git?ref=feature/extra-trusted-entities"`

Any help would be appreciated thanks!

**Comment by antonbabenko** [comment]
Should work like this - `source = "git::git@github.com:parmouraly/terraform-aws-atlantis.git?ref=feature/extra-trusted-entities"`

**Comment by parmouraly** [comment]
@antonbabenko are you waiting anything from me on this or can we merge please?

File: main.tf [hunk 1]
```diff
@@ -364,7 +364,7 @@ data "aws_iam_policy_document" "ecs_tasks" {
 
     principals {
       type        = "Service"
-      identifiers = ["ecs-tasks.amazonaws.com"]
+      identifiers = concat(["ecs-tasks.amazonaws.com"], var.trusted_principals)
```
Comment (antonbabenko) [inline main.tf:367]: It is better to have `compact(distinct(concat(...)))` to remove empty and repetitive values. Could you please update this?

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Looks good, a minor comment before I can merge.

File: main.tf (hunk 1 above)
Comment (parmouraly) [reply main.tf]: Done 👍

**Comment by antonbabenko** [comment]
Thanks!

[v2.27.0](https://github.com/terraform-aws-modules/terraform-aws-atlantis/releases/v2.27.0) has been just released.

**Pull request merged**